### ISSUE TITLE
Complex query input does not get parsed correctly to query params

### ISSUE THREAD

**Issue by Mysame** [troyanskiy/ngx-resource#135]
The following creates url the likes of;

api/query?filter=[Object object]

Expected:

api/query?
filter%5Blogic%5D=and
filter%5Bfilters%5D%5B0%5D%5Bfield%5D=name
filter%5Bfilters%5D%5B0%5D%5Boperator%5D=contains
filter%5Bfilters%5D%5B0%5D%5Bvalue%5D=bla

```
export class SearchParameters {
    skip?: number;
    take?: number;
    sort?: SortDescriptor[];
    filter?: FilterDescriptor[];
    search?: string;
}

export class SortDescriptor {
    dir: string;
    field: string;
}

export class FilterDescriptor {
    value?: string;
    field?: string;
    operator?: string;
    logic: string;
    filterDescriptors?: FilterDescriptor[];
}
```

**Comment by troyanskiy** [comment]
Hello @Mysame 
Thank you for the issue.
Yesterday I've discussed the topic with @jeremypele at PR #87 and he told that he will write PR to cover the case.

**Comment by jeremypele** [comment]
@Mysame Work in progress.. Shouldn't be long ;)

**Comment by jeremypele** [comment]
@troyanskiy PR available (and README update to avoid this kind of future issue) => https://github.com/troyanskiy/ngx-resource/pull/136

**Comment by troyanskiy** [comment]
New version of the lib has been published (3.5.0)
Thanks to @jeremypele !

I will close the issue.
Feel free to open.

Thank you!
